database/db.go

```diff
@@ -88,24 +86,41 @@ func New(ctx context.Context, creds database.DBCredentials, logger *logrus.Logge
 		return nil, err
 	}
 	return &DB{
-		DB:       db,
-		Skylinks: db.Collection(dbSkylinks),
-		Logger:   logger,
+		db,
+		logger,
 	}, nil
 }
 
+// Collection gets a handle for a collection with the given name configured with
+// the given CollectionOptions.
+func (db *DB) Collection(name string, opts ...*options.CollectionOptions) *mongo.Collection {
```

Comment:
f/u

This is better but not quite what I meant. The idea of the `DB` type is to completely abstract mongo away from the calling package. So it shouldn't return any mongo types or have passthrough methods.

Reply:
Coincidentally I have something similar in the abuse email scanner. I think that's a good example of a mongo abstraction. It has a mongo specific (generic) database file which contains mongo related methods, like ensuring database schema and stuff like that, and then you have a database abstraction specific to your use case, in my case that's called the `AbuseScannerDB` which exposes mutators like `FindX` `UpdateY` etc, that way you totally abstract mongo away and your database is only returning/receiving entities within your domain model.

Reply:
F/U https://github.com/SkynetLabs/malware-scanner/issues/10